test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPushCommandTest.cs

```diff
@@ -2078,6 +2084,339 @@ public void PushCommand_PushToServerV3_WithSymbols_ApiKey_SymbolApiKey_BothFromC
+                    configFilePath,
+                    configFileName,
+                    string.Format(
+                        CultureInfo.InvariantCulture,
+                        @"Not XML!"));
+
+                var args = new string[]
+                {
+                    Command_Push,
+                    packageFileName,
+                    Param_Source,
+                    repositoryKey,
+                    Param_ConfigFile,
+                    Path.Combine(configFilePath, configFileName)
+                };
+
+                var result = CommandRunner.Run(
+                                NuGetExePath,
+                                testDirectory,
+                                string.Join(" ", args),
+                                true);
+
+                // TODO: The bad config file is ignored, so I need to confirm this is expected behavior.
+                // This test needs to be updated before merging since it is failing 100%.
+                Assert.True(
+                   result.ExitCode != 0,
+                   "The run did not fail as desired. Simply got this output:" + result.Output);
+
+                var expectedError = string.Format(
+                                    CultureInfo.InvariantCulture,
+                                    "Invalid file '{0}'",
+                                    Path.Combine(CurrentDirectoryPath, configFileName).ToString());
+
+                Assert.True(
+                    result.Errors.Contains("NuGet.Config is not valid XML."),
+                    string.Format(
+                        CultureInfo.InvariantCulture,
+                        "Error was expected to contain: {0} \r\nActual error: {1}",
+                        expectedError, result.Errors));
+
+                Assert.False(
+                    File.Exists(Path.Combine(repositoryPath, packageFileName)),
+                    string.Format(
+                        CultureInfo.InvariantCulture,
+                        @"The package {0} was found in {1}",
+                        packageFileName, repositoryPath));
+            }
+        }
+
+        [Fact]
+        public void PushCommand_ConfigFile_ExpectedFormat()
+        {
+            var configFileName = "My.Config";
+            var repositoryKey = "MySource";
+
+            using (var testDirectory = TestDirectory.Create())
+            {
+                var repositoryPath = Path.Combine(testDirectory, "repository");
+                var configFilePath = Path.Combine(testDirectory, "config");
+                Directory.CreateDirectory(repositoryPath);
+                Directory.CreateDirectory(configFilePath);
+
+                var packageFileName = Util.CreateTestPackage("testPackage1", "1.1.0", testDirectory);
+
+                FileSystemUtils.CreateFile(
+                    configFilePath,
+                    configFileName,
+                    string.Format(
+                        CultureInfo.InvariantCulture,
```

Comment:
I dislike nested method calls, because it's harder to read (where is the end `)` for `string.Format`? Are there any additional parameters to `FileSystemUtils.CreateFile`, or just the 3? Also, when debugging, maybe I want to see the result of `string.Format` without stepping into the outer method call. If `string.Format` was a method from our own code, rather than the BCL (or if I disabled "Just My Code" in the debugger), then it's also harder to step into `FileSystemUtils.CreateFile` without stepping into the nested method call.

Therefore, I suggest extracting the `string.Format` into a variable, and then have a simple method call to `FileSystemUtils.CreateFile`.